### PR TITLE
Refactoring, Adding vnet and adding SC services (place holders)

### PR DESCRIPTION
**Pull request by eladiw**
1. Refactored - app services extracted to a different tf file, to be more readable
2. vnet and subnet created. all resources are now connected to the vnet/subnet
3. removed redundant postgres module
4. aligned names
5. all applications are now deployed (with dummy image - nginx, until actual services are ready)

Regarding the networking configuration:
1. 1 Vnet
2. 3 subnets 
    2.1 web_apps_subnet 10.0.1.0/24   -> for all the different web apps
    2.2 sql_subnet            10.0.2.0/24  -> for both sql servers (fhir + roles)
    2.3 kv_subnet             10.0.3.0/24   -> for the kv.
3. DNS zone per SQL:
    3.1 <resourcename>-fhir.database.windows.net
    3.2 <resourcename>-roles.database.windows.net
4. DNS zone for the kv
    4.1 <resourcename>.vault.azure.net

### CODE REVIEW

**Review by shajidissa — DISMISSED** [review]

File: sample_trial/definition.json [hunk 1]
```diff
@@ -14,5 +14,19 @@
     "config_server_service": {
         "image_name": "ghcr.io/ndph-arts/global-trial-config-mock",
         "image_tag": "latest"
+    },
+    "sc": {
```
Comment (balteravishay) [inline sample_trial/definition.json]: expend to spring-cloud

File: terraform/trial_rg/main.tf [hunk 2]
```diff
@@ -20,65 +20,95 @@ resource "azurerm_app_service_plan" "apps_service_plan" {
...
+  ]
 }
 
-# Audit storage
-module "trial_audit" {
-  source     = "./modules/audit"
-  trial_name = var.trial_name
-  rg_name    = azurerm_resource_group.trial_rg.name
+ # Add vnet integration to sql databases
+resource "azurerm_sql_virtual_network_rule" "sqlvnetrule" {
```
Comment (balteravishay) [inline terraform/trial_rg/main.tf]: this seems to correspond to service endpoint and not a private endpoint which is what i think we want here.

Comment (eladiw) [reply terraform/trial_rg/main.tf]: added private endpoint and private link.

**Review by shajidissa — APPROVED** [review]

**Review by balteravishay — APPROVED** [review]
lgtm

**Pull request merged**